### PR TITLE
LKE-591 cluster detail part II

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

I really didn't plan out a good way of breaking this down into chunks. Apologies for the size of this part. Part 2 includes:

- using live data
- enabling local state management, so the buttons and forms should all work normally
- updates to the Button component so the "remove" text can be dynamic
- adding an "editable" mode to the NodePoolDisplayTable
- enabling the Download kubeconfig button (uses same logic as the action menu on the landing page)

It does not include:

- Form submission logic, which will come in Part 3 (with API support to follow at an unspecified future date)
- a bunch of additional Redux boilerplate and services to support the form submission (part 3, maybe part 2.5 if needed)
- finished loading and error states for all stages of the flow (also part 3)
- Styles (Kayla will handle those for Part 4)
- Tests (Part 5)

## Type of Change
- Non breaking change ('update', 'change')

## Notes

The tags panel component "works", but the API doesn't support it yet so nothing is saved.

### CODE REVIEW

File: src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx [hunk 1]
```diff
@@ -53,21 +60,34 @@ export const NodePoolsDisplay: React.FunctionComponent<
...
+              types={types}
+              handleDelete={deletePool}
+              updatePool={updatePool}
+            />
           ) : (
-            <StaticPoolsDisplay pools={pools} types={types} />
+            <NodePoolDisplayTable
+              pools={pools}
+              types={types}
+              handleDelete={() => null}
```
Comment (johnwcallahan) [inline src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx]: Not sure if this is in scope of this PR, but it seems like clicking this button should take you to Edit mode.

Comment (Jskobos) [reply src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx]: We debated that, I wanted to hide the button unless you were in edit mode, we compromised on just disabling it. Will probably change after review.

File: src/features/Kubernetes/KubernetesClusterDetail/KubernetesClusterDetail.tsx [hunk 2]
```diff
@@ -93,22 +258,27 @@ export const KubernetesClusterDetail: React.FunctionComponent<
...
               types={typesData || []}
-              nodeCount={0}
-              addNodePool={() => null}
-              deleteNodePool={() => null}
-              handleTypeSelect={() => null}
-              updateNodeCount={() => null}
-              updatePool={() => null}
+              nodeCount={count}
+              addNodePool={handleAddNodePool}
+              deleteNodePool={() => null} // Not needed since we're not displaying the table from inside this component
```
Comment (martinmckenna) [inline src/features/Kubernetes/KubernetesClusterDetail/KubernetesClusterDetail.tsx]: if these are never being leveraged, they should be optional props, yeah?

Comment (Jskobos) [reply src/features/Kubernetes/KubernetesClusterDetail/KubernetesClusterDetail.tsx]: Yup, will fix

**Pull request merged**